src/linear_solver_routines/block_solver_helpers/gentlemans_house.jl

```diff
@@ -0,0 +1,155 @@
+# This is file from RLinearAlgebra.jl
+"""
+    GentData{S, M<:Matrix{S}, V<:Vector{S}}
+
+This is a preallocated data structure for solving least squares problems with 
+Gentleman's algorithm. This data structure is not exported and thus is not 
+designed to be manipulated by the user.
+# Fields
+- `A::M`, The matrix to be solved.
+- `B::M`, The matrix that rows of `A` are brought to have QR decomposition
+applied.
+- `R::UpperTriangular`, The upper triangular part of `B`.
+- `tab::SubArray`, the last column of `B` where the constant vector entries corresponding
+to the rows of `A` that were brought to `B` are stored.
+- `v::V`, a buffer vector.
```

Comment:
Needs a more precise description. Can reference LAPACK docs.

Reply:
I have added an additional statement directing them to the docs as well as a brief description of the purpose of the  vector.